server/datastore/mysql/software_installers.go

```diff
@@ -148,23 +149,31 @@ func (ds *Datastore) getOrGenerateSoftwareInstallerTitleID(ctx context.Context,
 	return titleID, nil
 }
 
-func (ds *Datastore) GetSoftwareInstallerMetadata(ctx context.Context, id uint) (*fleet.SoftwareInstaller, error) {
+func (ds *Datastore) GetSoftwareInstallerMetadata(ctx context.Context, id uint, includeTitle bool) (*fleet.SoftwareInstaller, error) {
```

Comment:
FWIW, I think it would be fine to always return that field and do without the boolean option.